### ISSUE TITLE
segmentation fault (core dumped)  dotnet --version

### ISSUE THREAD

**Issue by lucax88x** [dotnet/runtime#48128]
Suddenly, probably after an OS update, all dotnet commands  (even dotnet --version) are giving the SEGV error.

I am on Archlinux with kernel 5.10.11-arch1-1.

Shouldn't depend on the SDK, as I was already on 5.0.100 and worked for 3/4 months without any problem, tried to delete and download also 5.0.102 but problem persists.

this is the journal logs

<details>
  <summary>journalctl</summary>
  
systemd-coredump[123448]: Process 123429 (dotnet) of user 1000 dumped core.
                                                        
                                                        Stack trace of thread 123443:
                                                        #0  0x00007ff76dd83b0c n/a (libcoreclr.so + 0x12bb0c)
                                                        dotnet/sdk#1  0x00007ff76deace73 n/a (libcoreclr.so + 0x254e73)
                                                        dotnet/sdk#2  0x00007ff76deadb7b n/a (libcoreclr.so + 0x255b7b)
                                                        dotnet/sdk#3  0x00007ff76deae071 n/a (libcoreclr.so + 0x256071)
                                                        dotnet/sdk#4  0x00007ff76deae248 n/a (libcoreclr.so + 0x256248)
                                                        dotnet/sdk#5  0x00007ff76deae536 n/a (libcoreclr.so + 0x256536)
                                                        dotnet/sdk#6  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#7  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#8  0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123438:
                                                        #0  0x00007ff76ed326a2 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0 + 0xf6a2)
                                                        dotnet/sdk#1  0x00007ff76e13532b n/a (libcoreclr.so + 0x4dd32b)
                                                        dotnet/sdk#2  0x00007ff76e134fe1 n/a (libcoreclr.so + 0x4dcfe1)
                                                        dotnet/sdk#3  0x00007ff76e1399c2 n/a (libcoreclr.so + 0x4e19c2)
                                                        dotnet/sdk#4  0x00007ff76e139cd1 n/a (libcoreclr.so + 0x4e1cd1)
                                                        dotnet/sdk#5  0x00007ff76e027019 n/a (libcoreclr.so + 0x3cf019)
                                                        dotnet/sdk#6  0x00007ff76e026e85 n/a (libcoreclr.so + 0x3cee85)
                                                        dotnet/sdk#7  0x00007ff76e026bcd n/a (libcoreclr.so + 0x3cebcd)
                                                        dotnet/sdk#8  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#9  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#10 0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123433:
                                                        #0  0x00007ff76e90c46f __poll (libc.so.6 + 0xf546f)
                                                        dotnet/sdk#1  0x00007ff76e0300ee n/a (libcoreclr.so + 0x3d80ee)
                                                        dotnet/sdk#2  0x00007ff76de71be4 n/a (libcoreclr.so + 0x219be4)
                                                        dotnet/sdk#3  0x00007ff76de37b97 n/a (libcoreclr.so + 0x1dfb97)
                                                        dotnet/sdk#4  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#5  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#6  0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123431:
                                                        #0  0x00007ff76e90c46f __poll (libc.so.6 + 0xf546f)
                                                        dotnet/sdk#1  0x00007ff76e137670 n/a (libcoreclr.so + 0x4df670)
                                                        dotnet/sdk#2  0x00007ff76e136cd3 n/a (libcoreclr.so + 0x4decd3)
                                                        dotnet/sdk#3  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#4  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#5  0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123440:
                                                        #0  0x00007ff76ed329c8 pthread_cond_timedwait@@GLIBC_2.3.2 (libpthread.so.0 + 0>
                                                        dotnet/sdk#1  0x00007ff76e135316 n/a (libcoreclr.so + 0x4dd316)
                                                        dotnet/sdk#2  0x00007ff76e134fe1 n/a (libcoreclr.so + 0x4dcfe1)
                                                        dotnet/sdk#3  0x00007ff76e1399c2 n/a (libcoreclr.so + 0x4e19c2)
                                                        dotnet/sdk#4  0x00007ff76e139bf9 n/a (libcoreclr.so + 0x4e1bf9)
                                                        dotnet/sdk#5  0x00007ff76def019a n/a (libcoreclr.so + 0x29819a)
                                                        dotnet/sdk#6  0x00007ff76de5578f n/a (libcoreclr.so + 0x1fd78f)
                                                        dotnet/sdk#7  0x00007ff76de55947 n/a (libcoreclr.so + 0x1fd947)
                                                        dotnet/sdk#8  0x00007ff76ddd973a n/a (libcoreclr.so + 0x18173a)
                                                        dotnet/sdk#9  0x00007ff76ddd9ded n/a (libcoreclr.so + 0x181ded)
                                                        dotnet/sdk#10 0x00007ff76de55b38 n/a (libcoreclr.so + 0x1fdb38)
                                                        dotnet/sdk#11 0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#12 0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#13 0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123429:
                                                        #0  0x00007ff76ed33bcb __pthread_getspecific (libpthread.so.0 + 0x10bcb)
                                                        dotnet/sdk#1  0x00007ff76e130538 n/a (libcoreclr.so + 0x4d8538)
                                                        dotnet/sdk#2  0x00007ff76dd254af n/a (libcoreclr.so + 0xcd4af)
                                                        dotnet/sdk#3  0x00007ff76e0f5bab n/a (libcoreclr.so + 0x49dbab)
                                                        dotnet/sdk#4  0x00007ff76ddad864 n/a (libcoreclr.so + 0x155864)
                                                        dotnet/sdk#5  0x00007ff76dd8b1fe n/a (libcoreclr.so + 0x1331fe)
                                                        dotnet/sdk#6  0x00007ff76de95250 n/a (libcoreclr.so + 0x23d250)
                                                        dotnet/sdk#7  0x00007ff76de8cb91 n/a (libcoreclr.so + 0x234b91)
                                                        dotnet/sdk#8  0x00007ff76de9da10 n/a (libcoreclr.so + 0x245a10)
                                                        dotnet/sdk#9  0x00007ff76dd0ecd5 n/a (libcoreclr.so + 0xb6cd5)
                                                        dotnet/sdk#10 0x00007ff76dd0e748 n/a (libcoreclr.so + 0xb6748)
                                                        dotnet/sdk#11 0x00007ff76dd0faa8 n/a (libcoreclr.so + 0xb7aa8)
                                                        dotnet/sdk#12 0x00007ff76dd0b734 n/a (libcoreclr.so + 0xb3734)
                                                        dotnet/sdk#13 0x00007ff76dd0b5eb n/a (libcoreclr.so + 0xb35eb)
                                                        dotnet/sdk#14 0x00007ff76dd0d1d7 n/a (libcoreclr.so + 0xb51d7)
                                                        dotnet/sdk#15 0x00007ff76ddbc159 n/a (libcoreclr.so + 0x164159)
                                                        dotnet/sdk#16 0x00007ff76dd0d694 n/a (libcoreclr.so + 0xb5694)
                                                        dotnet/sdk#17 0x00007ff76de99b93 n/a (libcoreclr.so + 0x241b93)
                                                        dotnet/sdk#18 0x00007ff76dd01e55 n/a (libcoreclr.so + 0xa9e55)
                                                        dotnet/sdk#19 0x00007ff76dd01eb5 n/a (libcoreclr.so + 0xa9eb5)
                                                        dotnet/sdk#20 0x00007ff76dd0e7a6 n/a (libcoreclr.so + 0xb67a6)
                                                        dotnet/sdk#21 0x00007ff76dd0faa8 n/a (libcoreclr.so + 0xb7aa8)
                                                        dotnet/sdk#22 0x00007ff76dd0b734 n/a (libcoreclr.so + 0xb3734)
                                                        dotnet/sdk#23 0x00007ff76dd0ca37 n/a (libcoreclr.so + 0xb4a37)
                                                        dotnet/sdk#24 0x00007ff76dd0d174 n/a (libcoreclr.so + 0xb5174)
                                                        dotnet/sdk#25 0x00007ff76ddbc159 n/a (libcoreclr.so + 0x164159)
                                                        dotnet/sdk#26 0x00007ff76dd0d694 n/a (libcoreclr.so + 0xb5694)
                                                        dotnet/sdk#27 0x00007ff76de99b93 n/a (libcoreclr.so + 0x241b93)
                                                        dotnet/sdk#28 0x00007ff76dd01e55 n/a (libcoreclr.so + 0xa9e55)
                                                        dotnet/sdk#29 0x00007ff76dd01eb5 n/a (libcoreclr.so + 0xa9eb5)
                                                        dotnet/sdk#30 0x00007ff76dd0e7a6 n/a (libcoreclr.so + 0xb67a6)
                                                        dotnet/sdk#31 0x00007ff76dd0faa8 n/a (libcoreclr.so + 0xb7aa8)
                                                        dotnet/sdk#32 0x00007ff76dd0b734 n/a (libcoreclr.so + 0xb3734)
                                                        dotnet/sdk#33 0x00007ff76dd0b98f n/a (libcoreclr.so + 0xb398f)
                                                        dotnet/sdk#34 0x00007ff76dd01ead n/a (libcoreclr.so + 0xa9ead)
                                                        dotnet/sdk#35 0x00007ff76dd0e7a6 n/a (libcoreclr.so + 0xb67a6)
                                                        dotnet/sdk#36 0x00007ff76dd0faa8 n/a (libcoreclr.so + 0xb7aa8)
                                                        dotnet/sdk#37 0x00007ff76dd0b734 n/a (libcoreclr.so + 0xb3734)
                                                        dotnet/sdk#38 0x00007ff76dd0b5eb n/a (libcoreclr.so + 0xb35eb)
                                                        dotnet/sdk#39 0x00007ff76dd0d1d7 n/a (libcoreclr.so + 0xb51d7)
                                                        dotnet/sdk#40 0x00007ff76ddbc159 n/a (libcoreclr.so + 0x164159)
                                                        dotnet/sdk#41 0x00007ff76ddfbcc3 n/a (libcoreclr.so + 0x1a3cc3)
                                                        dotnet/sdk#42 0x00007ff76dd7b9df n/a (libcoreclr.so + 0x1239df)
                                                        dotnet/sdk#43 0x00007ff76dcf9545 n/a (libcoreclr.so + 0xa1545)
                                                        dotnet/sdk#44 0x00007ff76de00654 n/a (libcoreclr.so + 0x1a8654)
                                                        dotnet/sdk#45 0x00007ff76ddffb16 n/a (libcoreclr.so + 0x1a7b16)
                                                        dotnet/sdk#46 0x00007ff76ddae7e1 n/a (libcoreclr.so + 0x1567e1)
                                                        dotnet/sdk#47 0x00007ff76ddae8b7 n/a (libcoreclr.so + 0x1568b7)
                                                        dotnet/sdk#48 0x00007ff76ddae5ce n/a (libcoreclr.so + 0x1565ce)
                                                        dotnet/sdk#49 0x00007ff76dd15b75 n/a (libcoreclr.so + 0xbdb75)
                                                        dotnet/sdk#50 0x00007ff76ddb13cb n/a (libcoreclr.so + 0x1593cb)
                                                        dotnet/sdk#51 0x00007ff76ddb0e6a n/a (libcoreclr.so + 0x158e6a)
                                                        dotnet/sdk#52 0x00007ff76dfc29ac n/a (libcoreclr.so + 0x36a9ac)
                                                        dotnet/sdk#53 0x00007ff76dfc1db7 n/a (libcoreclr.so + 0x369db7)
                                                        dotnet/sdk#54 0x00007ff76de100b6 n/a (libcoreclr.so + 0x1b80b6)
                                                        dotnet/sdk#55 0x00007ff76de1032f n/a (libcoreclr.so + 0x1b832f)
                                                        dotnet/sdk#56 0x00007ff76dd944b4 n/a (libcoreclr.so + 0x13c4b4)
                                                        dotnet/sdk#57 0x00007ff76dd949ae n/a (libcoreclr.so + 0x13c9ae)
                                                        dotnet/sdk#58 0x00007ff76ddb2f3a n/a (libcoreclr.so + 0x15af3a)
                                                        dotnet/sdk#59 0x00007ff76ddb43b8 n/a (libcoreclr.so + 0x15c3b8)
                                                        dotnet/sdk#60 0x00007ff76dfc203b n/a (libcoreclr.so + 0x36a03b)
                                                        dotnet/sdk#61 0x00007ff6f4db096a n/a (n/a + 0x0)
                                                        dotnet/sdk#62 0x00007ff76dfc1db7 n/a (libcoreclr.so + 0x369db7)
                                                        dotnet/sdk#63 0x00007ff76de109eb n/a (libcoreclr.so + 0x1b89eb)
                                                        
                                                        Stack trace of thread 123436:
                                                        #0  0x00007ff76ed36774 __open (libpthread.so.0 + 0x13774)
                                                        dotnet/sdk#1  0x00007ff76e03067f n/a (libcoreclr.so + 0x3d867f)
                                                        dotnet/sdk#2  0x00007ff76e029738 n/a (libcoreclr.so + 0x3d1738)
                                                        dotnet/sdk#3  0x00007ff76e028879 n/a (libcoreclr.so + 0x3d0879)
                                                        dotnet/sdk#4  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#5  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#6  0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
                                                        
                                                        Stack trace of thread 123442:
                                                        #0  0x00007ff76ed329c8 pthread_cond_timedwait@@GLIBC_2.3.2 (libpthread.so.0 + 0>
                                                        dotnet/sdk#1  0x00007ff76e135316 n/a (libcoreclr.so + 0x4dd316)
                                                        dotnet/sdk#2  0x00007ff76e134fe1 n/a (libcoreclr.so + 0x4dcfe1)
                                                        dotnet/sdk#3  0x00007ff76e13a179 n/a (libcoreclr.so + 0x4e2179)
                                                        dotnet/sdk#4  0x00007ff76ddf8edd n/a (libcoreclr.so + 0x1a0edd)
                                                        dotnet/sdk#5  0x00007ff76ddf8db5 n/a (libcoreclr.so + 0x1a0db5)
                                                        dotnet/sdk#6  0x00007ff76e1406ae n/a (libcoreclr.so + 0x4e86ae)
                                                        dotnet/sdk#7  0x00007ff76ed2c3e9 start_thread (libpthread.so.0 + 0x93e9)
                                                        dotnet/sdk#8  0x00007ff76e917293 __clone (libc.so.6 + 0x100293)
</details>

**Comment by danmoseley** [comment]
@lucax88x can you share a dump file (either here, or by some private means)?

Can you try 3.1 and see if that works?

putting in VM area for now.

**Comment by lucax88x** [comment]
Unfortunately it's late now, after one days of debugging / installing / uninstalling I literally went for a full reinstall of the distro.

I tried different sdks (of dotnet 5, not 3) but they were all giving the same problem.

Pretty sure something was "corrupted" on the "distro" level, but that was the only application that was not working.. but I had to get my pc working so I took extreme measures.

One thing that I didn't think about, is that I had aliased "ps" to "procs" [link](https://github.com/dalance/procs) so maybe this was giving wrong handles to dotnet cli.. but I don't even know if CLI is even using "ps" for that.

Sorry about that..
